src/commissaire_http/authentication/__init__.py

```diff
@@ -143,20 +146,27 @@ def __call__(self, environ, start_response):
             result = authenticator.authenticate(environ, fake_start_response)
             # True means it was successful
             if result is True:
+                self.logger.debug('{} succeeded authentication.'.format(
+                    authenticator.__class__.__name__))
                 return self._app(environ, start_response)
-
-        # If the last result was False, do a generic Forbidden.
-        if result is False:
-            start_response(
-                '403 Forbidden', [('content-type', 'text/html')])
-            return [bytes('Forbidden', 'utf8')]
-        # Otherwise the plugin handled it's own start_response and
-        # return data. Pull from the fake_start_response and return
-        # the result from authenticate.
-        else:
-            start_response(
-                fake_start_response.body, fake_start_response.headers)
-            return result
+            # The plugin handled it's own start_response and
+            # return data. Pull from the fake_start_response and return
+            # the result from authenticate.
+            elif result is not False and fake_start_response.call_count > 0:
```

Comment:
nit: Would `isinstance(result, list)` be better than `result is not False` to weed out bad return types?